hls4ml/model/graph.py

```diff
@@ -318,46 +401,57 @@ class ModelGraph:
         outputs (list, optional):  The outputs to the model. If None, determined from layer_list
     """
 
-    def __init__(self, config, layer_list, inputs=None, outputs=None):
-        self.config = HLSConfig(config)
+    def __init__(self, config, inputs=None, outputs=None):
+        self.config = config
+        self.inputs = inputs
+        self.outputs = outputs
+        self.graph = OrderedDict()
+        self._applied_flows = []  # keep track of the applied flows
+        self.index = 0
+        self.output_vars = {}
+        self._top_function_lib = None
 
-        # keep track of the applied flows
-        self._applied_flows = []
+    @classmethod
+    def from_layer_list(cls, config_dict, layer_list, inputs=None, outputs=None):
+        def _find_output_variable_names(layer_list, layer_names):
```

Comment:
Just a personal preference - but I am not the biggest fan of functions inside of functions unless absolutely necessary. Is there a reason this function got moved?

Reply:
No other function uses it and it doesn't depend on `self`. If it was separate it would also have to be a `classmethod`, but it felt weird to have a class method intended only for internal use. I actually considered moving it to a separate utility file, but that also felt like too much given that it really is used only once.